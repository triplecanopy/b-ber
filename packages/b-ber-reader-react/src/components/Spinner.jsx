import React from 'react'
import classNames from 'classnames'

const Spinner = props => (
  <div
    className={classNames('spinner', {
      'spinner--visible': props.spinnerVisible,
    })}
  >
    <div className="spinner__detail" />
  </div>
)

export default Spinner
