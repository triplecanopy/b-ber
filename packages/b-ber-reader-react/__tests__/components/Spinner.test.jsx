import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { render } from '@testing-library/react'
import Spinner from '../../src/components/Spinner'

const mockStore = configureMockStore()

describe('Spinner', () => {
  test('renders the component', () => {
    let tree
    let store = mockStore({
      userInterface: {
        spinnerVisible: true,
      },
    })

    tree = render(
      <Provider store={store}>
        <Spinner />
      </Provider>
    )

    expect(tree.container).toMatchSnapshot()

    store = mockStore({
      userInterface: {
        spinnerVisible: false,
      },
    })

    tree = render(
      <Provider store={store}>
        <Spinner />
      </Provider>
    )

    expect(tree.container).toMatchSnapshot()
  })
})
