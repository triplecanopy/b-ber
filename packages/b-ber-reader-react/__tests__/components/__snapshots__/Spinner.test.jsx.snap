// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders the component 1`] = `
<div
  className="spinner spinner--visible"
>
  <div
    className="spinner__detail"
  />
</div>
`;

exports[`Spinner renders the component 2`] = `
<div
  className="spinner"
>
  <div
    className="spinner__detail"
  />
</div>
`;
